### PR TITLE
Completed smart list

### PR DESCRIPTION
**Pull request by edfloreshz**
This PR implements a smart list for completed tasks.

Screenshot_20230520_141858

It also clears task lists when they don't have more tasks to complete.

Screenshot_20230520_141828

### CODE REVIEW

File: src/widgets/content/helpers.rs [hunk 1]
```diff
@@ -209,19 +220,40 @@ pub async fn select_task_list(
...
+			fl!("list-empty").clone()
+		};
+
+	model.page_subtitle =
+		if matches!(model.parent_list, Some(SidebarList::Custom(_))) {
+			fl!("all-done-instructions").clone()
+		} else {
+			fl!("instructions").clone()
+		};
+
```
Comment (AaronErhardt) [inline src/widgets/content/helpers.rs:256]: `matches!(model.parent_list, Some(SidebarList::Custom(_)))` is used multiple times, you could just use a variable to store the result once to make the code more readable.

File: src/widgets/content/model.rs [hunk 2]
```diff
@@ -11,11 +11,13 @@ pub struct ContentModel {
...
 	pub task_entry: Controller<TaskInputModel>,
 	pub parent_list: Option<SidebarList>,
-	pub compact: bool,
 	pub icon: Option<String>,
 	pub title: String,
 	pub description: String,
 	pub smart: bool,
 	pub selected_task: Option<Task>,
 	pub show_task_details: bool,
+	pub page_icon: String,
```
Comment (AaronErhardt) [inline src/widgets/content/model.rs:20]: Is there a reason why this can't be `&'static str`?

Comment (edfloreshz) [reply src/widgets/content/model.rs]: No reason at all 👍🏼

File: src/widgets/content/helpers.rs (hunk 1 above)
Comment (edfloreshz) [reply src/widgets/content/helpers.rs]: I actually had a method that does that 🤣

**Comment by edfloreshz** [comment]
Thanks 🎉

**Pull request merged**